app/src/main/java/org/mozilla/fenix/components/toolbar/MenuPresenter.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.toolbar.BrowserToolbar
+
+class MenuPresenter(
+    private val menuToolbar: BrowserToolbar,
+    sessionManager: SessionManager
+) : SelectionAwareSessionObserver(sessionManager) {
+    /** Redraw the refresh/stop button */
```

Comment:
I'd add a `start` method here so we can observe a session by an optional sessionId, similar to [site permissions](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt#L84).

The reasoning behind that is, for Custom Tabs, those sessions will never be a selected session, so we have to tell our observers when we want to explicitly observe that kind of session.